src/assisted_installer_controller/operator_handler.go

```diff
@@ -145,14 +155,97 @@ func (handler ClusterServiceVersionHandler) IsInitialized() bool {
 	if err != nil {
 		return false
 	}
+	handler.log.Infof("Operator %s csv name is %s", handler.GetName(), csvName)
+	if _, err := handler.kc.GetCSV(handler.operator.Namespace, csvName); err != nil {
+		if apierrors.IsNotFound(err) {
+			handler.log.WithError(err).Warnf("CSV %s not exists for %s. "+
+				"Going to handle it with OLMEarlySetupBug fix", csvName, handler.GetName())
+
+			err = handler.handleOLMEarlySetupBug()
+			if err != nil {
+				handler.log.WithError(err).Warnf("Failed to handle OLMEarlySetupBug")
+			}
+		} else {
+			handler.log.WithError(err).Warnf("Failed to get csv %s not for %s. ", csvName, handler.GetName())
+		}
 
-	if csvName == "" {
 		return false
 	}
 
 	return true
 }
 
+// handleOLMEarlySetupBug cleans failed olm jobs as they will rerun by olm and
+// deletes install plans to force olm re-conciliation and job recreation.
+// Assisted Controller to take care of it once the cluster is up. The service
+// does this because subscriptions created too early in the installation
+// process might fail operator installation, and OLM will not retry installing
+// them.
+func (handler ClusterServiceVersionHandler) handleOLMEarlySetupBug() error {
+	handler.log.Infof("Check if there are failed olm jobs and delete them in case they exists")
+	err := handler.deleteFailedOlmJobs()
+	if err != nil {
+		handler.log.WithError(err).Warnf("Failed to delete olm jobs")
+		return err
+	}
+	// OLM for some reason doesn't reconcile the subscription becoming
+	// automatic, so we have to delete all the install plans to force OLM
```

Comment:
This comment no longer seems relevant

Reply:
it is relevant to , it doesn't reconcile it :(

Reply:
But now we delete the job because we want it to recreate, not because we changed it to automatic as the current comment implies

Reply:
fixed